junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/Node.java

```diff
@@ -159,6 +159,11 @@ public boolean isSkipped() {
 		public Optional<String> getReason() {
 			return Optional.ofNullable(this.reason);
 		}
+
+		@Override
+		public String toString() {
+			return skipped ? String.format("Skip [%s]", reason) : "Do not skip";
+		}
```

Comment:
This is not how we implement `toString()`.

Please rework in line with `org.junit.jupiter.api.extension.ConditionEvaluationResult.toString()`.